tools/topology/topology1/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -77,7 +77,8 @@ ifdef(`NOJACK', `',
 # PCM2 ---> volume ----> ALH 2 BE dailink 2
 ifdef(`MONO', `',
 `# PCM40 ---> volume ----> ALH 2 BE dailink 3')
-# PCM4 <--- volume <---- ALH 2 BE dailink 4
+ifdef(`NOMIC', `',
+`# PCM4 <--- volume <---- ALH 2 BE dailink 4')
```

Comment:
well, no this looks wrong. When the RT715 is not present, then we can still have a PCM4 pipeline that takes the data from ALH2 BE dailink 4

It's really not a 'NOMIC" solution, rather that the mics are handled in a different way.

Reply:
@plbossart I mean no dmic, not RT715 in this patch. On some platforms, there is no DMIC and only analog mic is used. There is no any codec connected to the BE dailink 4. I will change NOMIC to NODMIC.

BTW: And I found the comments are wrong. It should not be "ALH2 BE dailink4" but  "ALH2 BE $MIC_LINK" (more precisely, it should be  "ALH2 BE eval(MIC_LINK)" ).

Reply:
I still think you need to macros

NO_MIC_CODEC: tells that there is no RT715 on this skew
RT711_MIC_SUPPORT: tells that the local mics are connected to RT711.

you should really disable the mic pipeline only when the RT711_MIC_SUPPORT is not defined, otherwise the mic pipeline needs to rely on a capture port of the RT711 and the ALH,

Reply:
an acronym starting with "NOOB" doesn't sound very good in English, it's Internet slang for 'novice', 'newcomer' or 'inexperienced'. Consider a different term, such as NO_LOCAL_MICS or something.

I don't understand how you want to manage the case 2). You will need to use different branches to create the topology, and you need a variable to figure out if you are going to add the RT711 microphone path or the RT715 mic path. in existing solutions we use AMP_LINK as an integer, whereas here you need a true/false case.

The example I had in mind is the 'NO_JACK' case. We can have a similar case with NO_MIC_CODEC.

Reply:
I will use NO_LOCAL_MICS. Thanks for pointing it out.

When creating a pipeline for RT715 or RT711 local mic, we can both use:
```
ifdef(`NO_LOCAL_MIC', `' ,
`
# pipeline
PIPELINE_PCM_ADD(sof/pipe-highpass-switch-capture.m4,
        5, 4, 2, s32le,
        1000, 0, 0,
        48000, 48000, 48000)

# dai
DAI_ADD(sof/pipe-dai-capture.m4,
        5, ALH, eval(MIC_LINK * 256 + **PIN_ID**), `SDW'eval(MIC_LINK)`-Capture',
        PIPELINE_SINK_5, 2, s24le,
        1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)

# pcm
PCM_CAPTURE_ADD(Microphone, 4, PIPELINE_PCM_5)

# dai config
DAI_CONFIG(ALH, eval(MIC_LINK * 256 + **PIN_ID**), 4, `SDW'eval(MIC_LINK)`-Capture',
        ALH_CONFIG(ALH_CONFIG_DATA(ALH, eval(MIC_LINK * 256 + **PIN_ID**), 48000, 2)))
'
```

We don't need to care whether it is a rt715 codec or rt711 codec, do we?

Reply:
Thanks @libinyang, we don't need to know what the actual codec is in the topology file, all we see in ALH channels and PDI/port interfaces.